scripts/eqtl_hail_batch/conditional_analysis.py

```diff
@@ -0,0 +1,465 @@
+#!/usr/bin/env python3
+
+"""Perform conditional analysis on SNPs and expression residuals"""
+
+import os
+import hail as hl
+import hailtop.batch as hb
+import pandas as pd
+import statsmodels.api as sm
+import statsmodels.stats.multitest as multi
+from patsy import dmatrices  # pylint: disable=no-name-in-module
+from scipy.stats import spearmanr
+from cpg_utils.hail_batch import (
+    dataset_path,
+    output_path,
+    copy_common_env,
+    init_batch,
+    remote_tmpdir,
+)
+from cloudpathlib import AnyPath
+import click
+
+
+DEFAULT_DRIVER_MEMORY = '4G'
+DRIVER_IMAGE = os.getenv('CPG_DRIVER_IMAGE')
+assert DRIVER_IMAGE
+
+TOB_WGS = dataset_path('mt/v7.mt/')
+FREQ_TABLE = dataset_path('joint-calling/v7/variant_qc/frequencies.ht/', 'analysis')
+
+
+def get_number_of_scatters(residual_df, significant_snps_df):
```

Comment:
one general note, if you include Type inference data then pylint can help catch some sneaky bugs

```suggestion
def get_number_of_scatters(residual_df: pd.DataFrame, significant_snps_df: pd.DataFrame) -> int:
```

Reply:
That's a great tip! (I saw this in your and some other people's code earlier, and wasn't sure what exactly it did.) Thank you